crypto/fipsmodule/evp/evp_ctx.c

```diff
@@ -438,6 +439,50 @@ int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx) {
   return 1;
 }
 
+int EVP_PKEY_keygen_deterministic(EVP_PKEY_CTX *ctx,
+                                  EVP_PKEY **out_pkey,
+                                  const uint8_t *seed,
+                                  size_t *seed_len) {
+  // We have to avoid potential underlying services updating the indicator state,
+  // so we lock the state here.
+  FIPS_service_indicator_lock_state();
+  int ret = 0;
+  if (!ctx || !ctx->pmeth || !ctx->pmeth->keygen_deterministic) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
+    goto end;
+  }
+  if (ctx->operation != EVP_PKEY_OP_KEYGEN) {
+    OPENSSL_PUT_ERROR(EVP, EVP_R_OPERATON_NOT_INITIALIZED);
+    goto end;
+  }
+
+  if (!out_pkey) {
+    goto end;
+  }
+
+  if (!*out_pkey) {
+    *out_pkey = EVP_PKEY_new();
+    if (!*out_pkey) {
+      OPENSSL_PUT_ERROR(EVP, ERR_LIB_EVP);
+      goto end;
+    }
+  }
+
+  if (!ctx->pmeth->keygen_deterministic(ctx, *out_pkey, seed, seed_len)) {
+    EVP_PKEY_free(*out_pkey);
+    *out_pkey = NULL;
+    goto end;
+  }
+
+  ret = 1;
+end:
+  FIPS_service_indicator_unlock_state();
+  if(ret) {
+    EVP_PKEY_keygen_verify_service_indicator(*out_pkey);
+  }
```

Comment:
Should this be a separate function? Do the same requirements in EVP_PKEY_keygen_verify_service_indicator apply for randomized keygen and deterministic?

Reply:
Removed this check in [3578aee](https://github.com/aws/aws-lc/pull/1578/commits/3578aeef15c7532dc44a9604f8d9dd2e7793dae3) as the `EVP_PKEY_keygen_verify_service_indicator` function cannot yet handle KEMs. It is also unknown at this stage if the de-randomized APIs for ML-KEM will be part of the FIPS checks or not, as this is still upcoming work in FIPS 203 IPD v2 expected soon.